build/build_test.go

```diff
@@ -218,22 +212,199 @@ func TestOverlayAugmentation(t *testing.T) {
+				type (
+					Foo struct {}
+					bar interface{}
+					bob int
+				)`,
+			want: ``,
+			expInfo: map[string]overrideInfo{
+				`Foo`: {purgeMethods: true},
+				`bar`: {purgeMethods: true},
+				`bob`: {purgeMethods: true},
+			},
+		}, {
+			desc: `purge part of type decl`,
+			src: `type (
+					Foo struct {}
+
+					//gopherjs:purge
+					bar interface{}
+
+					//gopherjs:purge
+					bob int
+				)`,
+			want: `type (
+					Foo struct {}
+				)`,
+			expInfo: map[string]overrideInfo{
+				`Foo`: {},
+				`bar`: {purgeMethods: true},
+				`bob`: {purgeMethods: true},
+			},
+		}, {
+			desc: `purge all of a type decl`,
+			src: `type (
+					//gopherjs:purge
+					Foo struct {}
+				)`,
+			want: ``,
+			expInfo: map[string]overrideInfo{
+				`Foo`: {purgeMethods: true},
+			},
+		}, {
+			desc: `remove and purge values`,
+			src: `import "time"
+
+				const (
+					foo = 42
+					//gopherjs:purge
+					bar = "gopherjs"
+				)
+
+				//gopherjs:purge
+				var now = time.Now`,
+			want: `const (
+					foo = 42
+				)`,
+			expInfo: map[string]overrideInfo{
+				`foo`: {},
+				`bar`: {},
+				`now`: {},
+			},
+		}, {
+			desc: `purge all value names`,
+			src: `//gopherjs:purge
+				var foo, bar int
+
+				//gopherjs:purge
+				const bob, sal = 12, 42`,
+			want: ``,
+			expInfo: map[string]overrideInfo{
+				`foo`: {},
+				`bar`: {},
+				`bob`: {},
+				`sal`: {},
+			},
+		}, {
+			desc: `imports not confused by local variables`,
+			src: `import (
+					"cmp"
+					"time"
+				)
+
+				//gopherjs:purge
+				func Sort[S ~[]E, E cmp.Ordered](x S) {}
+
+				func SecondsSince(start time.Time) int {
+					cmp := time.Now().Sub(start)
+					return int(cmp.Second())
+				}`,
+			want: `import (
+					"time"
+				)
+
+				func SecondsSince(start time.Time) int {
+					cmp := time.Now().Sub(start)
+					return int(cmp.Second())
+				}`,
+			expInfo: map[string]overrideInfo{
+				`Sort`:         {},
+				`SecondsSince`: {},
+			},
+		}, {
+			desc: `purge generics`,
+			src: `import "cmp"
+
+				//gopherjs:purge
+				type Pointer[T any] struct {}
+
+				//gopherjs:purge
+				func Sort[S ~[]E, E cmp.Ordered](x S) {}
+
+				// stub for "func Equal[S ~[]E, E any](s1, s2 S) bool"
+				func Equal() {}`,
+			want: `// stub for "func Equal[S ~[]E, E any](s1, s2 S) bool"
+				func Equal() {}`,
+			expInfo: map[string]overrideInfo{
+				`Pointer`: {purgeMethods: true},
+				`Sort`:    {},
+				`Equal`:   {},
+			},
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.desc, func(t *testing.T) {
-			pkgName := "package testpackage\n\n"
+			const pkgName = "package testpackage\n\n"
+			if test.noCodeChange {
+				test.want = test.src
+			}
+
 			fsetSrc := token.NewFileSet()
 			fileSrc := srctesting.Parse(t, fsetSrc, pkgName+test.src)
 
 			overrides := map[string]overrideInfo{}
 			augmentOverlayFile(fileSrc, overrides)
+			pruneImports(fileSrc)
+
+			buf := &bytes.Buffer{}
+			_ = printer.Fprint(buf, fsetSrc, fileSrc)
+			got := buf.String()
```

Comment:
Here and below, I think `srctesting.Format()` can be used instead.

Reply:
Ooh, yes it can. I didn't see/remember that was available. I'll change to use that. Thanks

Reply:
Fixed